app/javascript/components/project-submissions/components/__tests__/flag-form.test.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import {
+  fireEvent, render, screen, act,
+} from '@testing-library/react';
+import FlagForm from '../flag-form';
+
+describe('FlagForm', () => {
+  let onSubmit;
+
+  beforeEach(() => {
+    onSubmit = jest.fn();
+  });
+
+  describe('when the user is not signed in', () => {
+    test('prompts the user to sign in', () => {
+      render((
+        <FlagForm
+          userId={null}
+          onSubmit={onSubmit}
+          submission={{ id: 47 }}
+        />
+      ));
+
+      expect(screen.getByRole('link', { name: /sign in/i })).toBeInTheDocument();
+    });
+  });
+
+  test('has the submission id in a hidden input', () => {
+    render(
+      <FlagForm
+        userId={43}
+        onSubmit={onSubmit}
+        submission={{ id: 47 }}
+      />,
+    );
+
+    expect(screen.getByDisplayValue('47', { hidden: true })).toBeInTheDocument();
+  });
```

Comment:
I don't think this test is necessary. We want to limit testing to the component's interactions, as they are presented to the user.

Reply:
Got it, makes sense. Removed!